File: JavaScript/217_ContainsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const containsDuplicate = function(nums) {
    const hashSet = new Set();

    for (let i = 0; i < nums.length; i++) {
        if (hashSet.has(nums[i])) {
            return true;
        }
        hashSet.add(nums[i]);
    }

    return false;
};

const testCases = [
    {
        nums: [1,2,3,1],
        expected: true
    },
    {
        nums: [1,2,3,4],
        expected: false
    },
    {
        nums: [1,1,1,3,3,4,3,2,4,2],
        expected: true
    }
]

testCases.forEach((testCase, i) => {
    const output = containsDuplicate(testCase.nums);
    if (JSON.stringify(output) === JSON.stringify(testCase.expected)) {
        console.log(`Test Case ${i}: Passed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    } else {
        console.log(`Test Case ${i}: Failed. Expected: ${JSON.stringify(testCase.expected)}, Output: ${JSON.stringify(output)} `)
    }
});

File: CSharp/217_ContainsDuplicate.cs
namespace LeetCodeSolutions {
    public static class ContainsDuplicate {
        public static bool Solution(int[] nums) {
            HashSet<int> hashSet = new HashSet<int>();

            foreach (int num in nums) {
                if (hashSet.Contains(num)) {
                    return true;
                }
                hashSet.Add(num);
            }
            return false;
        }

        public static void RunTests() {
            Console.WriteLine("Running Contains Duplicate");

            var testCases = new List<(int[] nums, bool expected)> {
                (new int[] {1, 2, 3, 1}, true),
                (new int[] {1, 2, 3, 4}, false),
                (new int[] {1, 1, 1, 3, 3, 4, 3, 2, 4, 2}, true)
            };

            foreach (var (nums, expected) in testCases) {
                try {
                    var result = Solution(nums);
                    Console.WriteLine(
                        $"Test Passed: Input: [{string.Join(", ", nums)}], Expected {expected}, Result: {result}" 
                    );
                } catch (Exception ex) {
                    Console.WriteLine($"Test Failed: {ex.Message}");
                }
            }
        }
    }
}

